File: study/lua/src/app/ui/component/CommonHeroCountryFlag.lua

local CommonHeroCountryFlag = class("CommonHeroCountryFlag")
local TypeConvertHelper = require("app.utils.TypeConvertHelper")

local EXPORTED_METHODS = {
    "updateUI",
}

function CommonHeroCountryFlag:ctor()
	self._target = nil
end

function CommonHeroCountryFlag:_init()
	self._imageColor = ccui.Helper:seekNodeByName(self._target, "ImageColor")
end

function CommonHeroCountryFlag:bind(target)
	self._target = target
    self:_init()
    cc.setmethods(target, self, EXPORTED_METHODS)
end

function CommonHeroCountryFlag:unbind(target)
    cc.unsetmethods(target, EXPORTED_METHODS)
end

function CommonHeroCountryFlag:updateUI(type, value, limitLevel, limitRedLevel)
	local param = TypeConvertHelper.convert(type, value, nil, nil, limitLevel, limitRedLevel)
	local color = param.color
	self._imageColor:loadTexture(Path.getBackground("img_light_0"..color, ".png"))
end

return CommonHeroCountryFlag

File: study/ts/script/ui/component/CommonHeroCountryFlag.ts
import { TypeConvertHelper } from "../../utils/TypeConvertHelper";
import CommonUI from "./CommonUI";
import { Path } from "../../utils/Path";

const { ccclass, property } = cc._decorator;

@ccclass
export default class CommonHeroCountryFlag extends cc.Component {

    @property({
        type: cc.Sprite,
        visible: true
    })
    _imageColor: cc.Sprite = null;

    updateUI(type, value, limitLevel, limitRedLevel): void {
        var param = TypeConvertHelper.convert(type, value, null, limitLevel, limitRedLevel);
        if (param && param.color) {
            this.node.active = true;
            var color = param.color;
            this._imageColor.addComponent(CommonUI).loadTexture(Path.getBackground('img_light_0' + color, '.png'));
        }else {
            this.node.active = false;
        }
    }

}